server.js

```diff
@@ -88,6 +88,7 @@ if (config.blockedAtWarnEnable) {
 const app = express();
 app.set('views', path.join(__dirname, 'pages'));
 app.set('view engine', 'ejs');
+app.set('trust proxy', true);
```

Comment:
Won’t this enable any client to spoof a CBTF IP by setting `X-Forwarded-For` to a CBTF address?

That is, if a client sends a request with `X-Forwarded-For = [Some-CBTF-IP]`, Apache will add the true client IP to the right-hand end of the `X-Forwarded-For` list, but `trust proxy` will make Express trust the first (left-hand) entry (the spoofed entry set by the client).

I think maybe we want `app.set(‘trust proxy’, ‘localhost’)`, but I haven’t tested it (or thought very hard about it).

An alternative would be to add a config variable and do `app.set(‘trust proxy’, config.trustProxyHops)`, where for us `config.trustProxyHops = 1`. I haven’t tested this either.

Reply:
That's fair. PS uses `app.set('trust proxy', 'loopback')` which should be safe here too?

Reply:
Maybe? What does that do exactly?